pkg/expr/expr_types.go

```diff
@@ -404,6 +432,85 @@ func CastToKV(val interface{}) (out map[string]string, err error) {
 	}
 }
 
+func (t *KV) Each(fn func(k string, v TypedValue) error) (err error) {
+	if t == nil || t.value == nil {
+		return
+	}
+
+	for k, v := range t.value {
+		if err = fn(k, Must(NewString(v))); err != nil {
```

Comment:
Must will panic. You use it when it's ok to panic (and it's not in most cases).
You return error from Each anyway, so better to try and do NewString, and return if it yields an error.